include/picongpu/fields/background/templates/twtstight/BField.tpp

```diff
@@ -262,15 +242,12 @@ namespace picongpu
                 /* To avoid underflows in computation, fields are set to zero
                  * before and after the respective TWTS pulse envelope.
                  */
-                if(math::abs(y - z * math::tan(phiT / float_T(2.0)) - (cspeed * t)) > (numSigmas * tauG * cspeed))
+                if(math::abs(y - z * tanAlpha - (beta0 * cspeed * t)) > (numSigmas * tauG * cspeed))
                     return float_T(0.0);
 
                 /* Calculating shortcuts for speeding up field calculation */
```

Comment:
I think you're perfectly fine to assume that the compiler would be capable of figuring this out for itself and will happily ignore if these are present or not, using its own heuristics instead. The only benefit of these could be simplifying formulae later on. You can decide for yourself if for any or all of these this purpose is fulfilled.

Reply:
Was this comment meant for a different line? Without this if-statement, the code produces NaNs due to underflow. We already were severely bitten by this. The function of this statement corresponds to the "laser window" of a standard Gaussian laser.

Reply:
Sorry, always forget how badly GH renders comments on a single line. This was supposed to be a comment on "Calculating shortcuts for speeding up field calculation".